internal/dockercompose/client.go

```diff
@@ -118,13 +120,17 @@ func (c *cmdDCClient) Down(ctx context.Context, configPaths []string, stdout, st
 }
 
 func (c *cmdDCClient) StreamLogs(ctx context.Context, configPaths []string, serviceName model.TargetName) (io.ReadCloser, error) {
-	// TODO(maia): --since time
-	// (may need to implement with `docker log <cID>` instead since `d-c log` doesn't support `--since`
 	var args []string
 	for _, config := range configPaths {
 		args = append(args, "-f", config)
 	}
-	args = append(args, "logs", "--no-color", "-f", serviceName.String())
+	// NOTE: we can't practically remove "--no-color" due to the way that Docker Compose formats colorful lines; it
+	// 		 will wrap the entire line (including the \n) with the color codes, so you end up with something like:
+	//			\u001b[36mmyproject_my-container_1 exited with code 0\n\u001b[0m
+	// 		 where the ANSI reset (\u001b[0m) is _AFTER_ the \n, which doesn't play nice with our log segment logic
+	// 		 under some conditions - adding a final \n after stdout is closed would probably be sufficient given the
+	// 		 current pattern of how Compose colorizes stuff, but it's really not worth the headache to find out
+	args = append(args, "logs", "--no-color", "--no-log-prefix", "--timestamps", "--follow", serviceName.String())
```

Comment:
i wonder if we should give up on trying to get the logs from `docker-compose`...instead, we'd get the containers ID from docker-compose, then use the docker client logs API to stream logs

(i would generally expect the docker logs API to be way more fully baked, since that's what CRI uses -- and it'd look a lot closer to our k8s log streaming code)

Reply:
I _ALMOST_ did that - we already have the container ID in state, so it'd be very straightforward to use the Docker client logs API.

My decision for not was
   A. mistakenly thinking this wasn't going to be a huge can of worms until too late
   B. in theory, the new Compose V2 lets you `docker-compose up` against Azure ACI/AWS ECS, so the CLI will handle using appropriate cloud API for operations [that said, Live Update won't work in its current form since it uses Docker API directly vs shelling out to `docker-compose cp` - so not sure if this even a use case we want to support]

But, yeah, I think how we approach Docker Compose might be interesting to re-visit when we do the API migration for it. At the very least, if we pull in the core of Compose V2 as a dependency now that it's in Go, we can avoid all the pitfalls of shelling out via CLI. (Maybe it could become its own operator extension vs being built-in to keep dependencies under control?)

Reply:
@milas For the record, I just tested this fix, and the need for docker-compose 1.28.0+ (because of `--no-log-prefix`) should be explicitly noted somewhere - I spent 20 minutes investigating why log-watching processes are still restarting.

Reply:
@mouchar Good catch - I have added that to the release notes. We currently don't have version detection logic but have considered adding it due to Compose v2 having known bugs - I'll make note of this as a case to detect as well.